### ISSUE TITLE
Wrong Julia version started on remote computer w/Juliaup

### ISSUE THREAD

**Issue by BioTurboNick** [julia-vscode/julia-vscode#3259]
I'm using VSCode to connect to Linux on an AWS server. I am using Juliaup to set the Julia version. I'm trying to compare profiles bof my code between Julia versions.

I had Juliaup set to the RC channel, and when I open the Julia REPL on my remote-connected VSCode, it loaded 1.9-rc1. Then I went to my SSH connection and set Juliaup to Release. Then I closed the VSCode terminal and tried to reopen it. But it still loaded 1.9-rc1. I confirmed that from the shell, 1.8.5 loads. And of course loading Julia from the shell even in VSCode, I can't use VSCodeServer tools like `@profview`.

What is locking the terminal to 1.9 here? How can I work around this?

**Comment by BioTurboNick** [comment]
Workaround: Discovered the connect external REPL tool, thankfully.

**Comment by pfitzseb** [comment]
Can you switch to the Remote tab in the VS Code settings and check the `julia.executablePath` value there?

**Comment by BioTurboNick** [comment]
It's blank

**Comment by BioTurboNick** [comment]
Is it possible it stores the absolute path `julia` maps to on first run, rather than use the shortcut each time? Restarting VS Code and reconnecting to the remote computer seemed to update it to the current default for `juliaup`.

**Comment by pfitzseb** [comment]
No, I don't think that should be possible.

**Comment by BioTurboNick** [comment]
Just saw the same thing on my computer. Upgraded from rc2 to rc3 with Juliaup, but VSCode kept trying to launch rc2 (and failing, because it was removed).

Closing and reopening VSCode fixed it.

**Comment by MilesCranmer** [comment]
Same issue here. Despite having Julia installed in `/dev/shm/.julia`, starting the REPL causes VSCode to try to download juliaup and install to `~/.julia`. Not sure why.

**Comment by BioTurboNick** [comment]
Might be a different issue then - here I'm already using Juliaup, but it seems to look up the `juliaup default` one time and then sticks with that, even if I later manually change the default while VSCode is running.

**Comment by davidanthoff** [comment]
Yeah, we query Juliaup once at startup for versions, and then never update that list. I have some ideas how to fix this.
